src/pages/my_page/user_profile/MyProfile.tsx

```diff
@@ -1,46 +1,167 @@
+  const [editPreferredNames, setEditPreferredNames] = useState(false);
+
+  const initialValues: CristinPersonFormData = {
+    preferredFirstName: personPreferredFirstName ? personPreferredFirstName : firstName,
+    preferredLastName: personPreferredLastName ? personPreferredLastName : lastName,
+  };
+
+  const updatePerson = async (values: CristinPersonFormData) => {
+    if (user.cristinId) {
+      values.preferredFirstName = values.preferredFirstName?.trim();
+      values.preferredLastName = values.preferredLastName?.trim();
+      if (values.preferredFirstName === '') {
+        values.preferredFirstName = null;
+      }
+      if (values.preferredLastName === '') {
+        values.preferredLastName = null;
+      }
+      const updatePersonResponse = await updateCristinPerson(user.cristinId, values);
+      if (isErrorStatus(updatePersonResponse.status)) {
+        dispatch(setNotification({ message: t('feedback.error.update_person'), variant: 'error' }));
+      } else if (isSuccessStatus(updatePersonResponse.status)) {
+        dispatch(setNotification({ message: t('feedback.success.update_person'), variant: 'success' }));
+        setEditPreferredNames(false);
+        refetchPerson();
+      }
+    }
+  };
+
   return (
     <>
       <Helmet>
         <title>{t('my_page.my_profile.user_profile')}</title>
       </Helmet>
-      <Box sx={{ display: 'grid', gridTemplateColumns: '3fr 1fr', gap: '1rem' }}>
+      <Box
+        sx={{
+          display: 'grid',
+          gridTemplateColumns: {
+            xs: '1fr',
+            md: '3fr 1fr',
+          },
+          gap: '1rem',
+          gridTemplateAreas: {
+            xs: '"user-profile" "research-profile"',
+            md: '"user-profile research-profile" ',
+          },
+        }}>
         <BackgroundDiv
           sx={{
             display: 'grid',
-            columnGap: '3rem',
             gridTemplateAreas: {
               xs: '"primary-info" "roles"',
-              md: '"roles primary-info"',
+              md: '"roles" "primary-info"',
             },
-            gridTemplateColumns: { xs: '1fr', md: '1fr 2fr' },
+            bgcolor: 'info.light',
+            gridArea: 'user-profile',
           }}>
-          <Box sx={{ gridArea: 'roles' }}>
-            <UserRoles user={user} />
-          </Box>
-
-          <Box sx={{ display: 'grid', gridArea: 'primary-info', gap: '1rem' }}>
-            <UserInfo user={user} />
-            <Divider />
-            <UserOrcid user={user} />
+          <Box sx={{ display: 'flex', flexDirection: 'column', gap: '1.5rem' }}>
+            <Typography variant="h2">{t('my_page.my_profile.heading.personalia')}</Typography>
+            {isLoadingPerson && !person ? (
+              <PageSpinner />
+            ) : (
+              <>
+                <Typography>{t('my_page.my_profile.user_profile_description')}</Typography>
+                <Formik initialValues={initialValues} onSubmit={updatePerson} enableReinitialize>
```

Comment:
Har vært litt frem og tilbake her også :D men kan godt verifisere om `enableReinitialize` trengs eller ikke. Hvis den trengs vil det sikkert bli trøbbel når man lagrer nye verdier, f.eks. ved å fjerne preferred name
```suggestion
                <Formik initialValues={initialValues} onSubmit={updatePerson}>
```